ios/FluentUI/HUD/HUDView.swift

```diff
@@ -38,81 +36,102 @@ class HUDView: UIView {
     }
 
     public required init?(coder aDecoder: NSCoder) {
         preconditionFailure("init(coder:) has not been implemented")
     }
 
-    open override func layoutSubviews() {
-        super.layoutSubviews()
-
-        indicatorView.sizeToFit()
-        indicatorView.frame = CGRect(
-            x: UIScreen.main.roundToDevicePixels((bounds.width - indicatorView.frame.width) / 2),
-            y: UIScreen.main.roundToDevicePixels(label.isHidden ? (bounds.height - indicatorView.frame.height) / 2 : Constants.paddingVerticalWithLabel),
-            width: indicatorView.frame.width,
-            height: indicatorView.frame.height
-        )
-
-        let labelSize = label.sizeThatFits(CGSize(width: Constants.labelMaxWidth, height: .greatestFiniteMagnitude))
-        label.frame = CGRect(
-            x: UIScreen.main.roundToDevicePixels((bounds.width - labelSize.width) / 2),
-            y: indicatorView.frame.maxY + Constants.labelMarginTop,
-            width: labelSize.width,
-            height: labelSize.height
-        )
-    }
-
     open override func sizeThatFits(_ size: CGSize) -> CGSize {
-        let indicatorViewSize = indicatorView.sizeThatFits(size)
-
         if label.isHidden {
-            let sideSize = max(Constants.minSideSizeWithoutLabel, indicatorViewSize.width + 2 * Constants.paddingHorizontalWithoutLabel)
-            return CGSize(width: sideSize, height: sideSize)
+            return CGSize(width: Constants.minWidth, height: Constants.minWidth)
         } else {
-            let labelSize = label.sizeThatFits(CGSize(width: Constants.labelMaxWidth, height: .greatestFiniteMagnitude))
-            let fittingWidth = max(indicatorViewSize.width, labelSize.width) + 2 * Constants.paddingHorizontalWithLabel
-            let fittingHeight = Constants.paddingVerticalWithLabel + indicatorViewSize.height + Constants.labelMarginTop + labelSize.height + Constants.paddingVerticalWithLabel
-
-            return CGSize(
-                width: min(Constants.maxWidth, max(Constants.minSideSizeWithLabel, fittingWidth)),
-                height: max(Constants.minSideSizeWithLabel, fittingHeight)
-            )
+            let activitySize = ActivityIndicatorViewSize.xLarge.sideSize
+            let maxWidth = traitCollection.preferredContentSizeCategory > .large ? Constants.maxWidthInLargerContent : Constants.maxWidth
+
+            let labelSize = label.sizeThatFits(CGSize(width: maxWidth - 2 * Constants.paddingHorizontal, height: .greatestFiniteMagnitude))
+
+            let fittingWidth = max(activitySize, labelSize.width) + 2 * Constants.paddingHorizontal
+            let fittingHeight = activitySize + Constants.labelMarginTop + labelSize.height + (2 * Constants.paddingVertical)
+
+            var suggestedSize = max(fittingWidth, fittingHeight)
+            suggestedSize = max(ceil(suggestedSize), Constants.minWidth)
+            suggestedSize = min(maxWidth, suggestedSize)
+
+            // make sure HUD is always a square
```

Comment:
very minor: might move this comment up before line 129 since it explains the contents of that block as well